SalesforceCommerceCloud/pwa-kit
# [Storefront Preview] SCAPI cache breaker for older v2 PWA Kit (@W-15214446@)

Porting the changes from #1701. Unfortunately, while the solution/concept is still the same, the implementation had to be different. 

In the older V2 PWA Kit:
- The `StorefrontPreview` component does not have access to the commerce api clients because they're not in the same package
- The `Proxy` mechanism has already been used. So we need to add the cache breaker using the existing mechanism (see `willSendRequest`)

Todos:
- [ ] Write some documentation for existing projects to consume the update?
- [x] Bump up version number? To release as a patch.

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

https://runtime.commercecloud.com/mobify/blitz/will/preview?path=/global/en-GB/product/25695327M?color%3DJJI15XX%26size%3D9SM%26pid%3D701644153083M

1. Note that the product price should be the regular price
2. In the Preview form, enter tester=yes as the custom qualifier
3. Click on the Preview button
4. The price is immediately updated with the discounted promo price.

Note: to verify that the environment is actually using a bundle for the _older_ PWA Kit, you can add an item to the cart, and that should result in an OCAPI call (not SCAPI).

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
